### PR TITLE
polish dist unit test code

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/dist_se_resnext.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def net(self, input, class_dim=1000):
 
         pool = fluid.layers.pool2d(
             input=conv, pool_size=7, pool_type='avg', global_pooling=True)
-        drop = fluid.layers.dropout(x=pool, dropout_prob=0.2)
+        drop = fluid.layers.dropout(x=pool, dropout_prob=0.2, seed=1)
```
Comment (chengduoZH) [inline python/paddle/fluid/tests/unittests/dist_se_resnext.py]: `main_program.random_seed` can set the seed of `layers.dropout` now.

https://github.com/PaddlePaddle/Paddle/blob/3300a532aa05ada96fde9fd908b562b3f32da6d9/python/paddle/fluid/layers/nn.py#L953-L954

**Review by chengduoZH — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/dist_se_resnext.py [hunk 2]
```diff
@@ -287,7 +289,7 @@ def run_trainer(self, place, endpoints, trainer_id, trainers, is_dist=True):
         else:
             trainer_prog = fluid.default_main_program()
 
-        startup_exe = fluid.Executor(place)
+        startup_exe = fluid.Executor(fluid.CUDAPlace(0))
```
Comment (chengduoZH) [inline python/paddle/fluid/tests/unittests/dist_se_resnext.py]: Does it only support GPU?

File: python/paddle/fluid/tests/unittests/dist_se_resnext.py [hunk 3]
```diff
@@ -302,12 +304,19 @@ def run_trainer(self, place, endpoints, trainer_id, trainers, is_dist=True):
         ]
 
         feeder = fluid.DataFeeder(feed_var_list, place)
-        reader_generator = train_reader()
-        first_loss, = exe.run(fetch_list=[avg_cost.name])
+        reader_generator = test_reader()
+
+        data = next(reader_generator)
+        first_loss, = exe.run(fetch_list=[avg_cost.name],
+                              feed=feeder.feed(data))
```
Comment (chengduoZH) [inline python/paddle/fluid/tests/unittests/dist_se_resnext.py:311]: line 309~311 can be simpled.

**Review by chengduoZH — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/dist_se_resnext.py (hunk 2 above)
Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/dist_se_resnext.py]: Done.

File: python/paddle/fluid/tests/unittests/dist_se_resnext.py (hunk 3 above)
Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/dist_se_resnext.py:311]: will refine later.

**Review by chengduoZH — APPROVED** [review]

File: paddle/fluid/operators/read_op.cc [hunk 4]
```diff
@@ -65,6 +66,12 @@ class ReadOp : public framework::OperatorBase {
             .GetMutable<framework::ReaderHolder>();
     std::vector<std::string> out_arg_names = Outputs("Out");
     std::vector<framework::LoDTensor> ins;
+
+    // For profiling
+    platform::DeviceContextPool& pool = platform::DeviceContextPool::Instance();
+    auto& ctx = *pool.Get(dev_place);
+    platform::RecordEvent record_event(Type(), &ctx);
```
Comment (panyx0718) [inline paddle/fluid/operators/read_op.cc:73]: if read_op runs on CPU, ctx can be nullptr

**Pull request merged**